src/Event/DefaultRoleEvent.php

```diff
@@ -131,26 +156,35 @@ public function getIterator() {
   }
 
   /**
+   * {@inheritdoc}
+   */
+  public function reset() {
+    $this->roles = [];
+  }
+
+  /**
    * Validates a role that is about to be set or added.
    *
-   * @param string $name
-   *   The name of the role to add or set.
    * @param array $properties
    *   The role properties to validate.
    *
    * @throws \InvalidArgumentException
    *   Thrown when the role name is empty, the 'label' property is missing, or
    *   the 'role_type' property is invalid.
    */
-  protected function validate($name, $properties) {
-    if (empty($name)) {
+  protected function validate($properties) {
+    if (empty($properties['name'])) {
       throw new \InvalidArgumentException('Role name is required.');
     }
 
     if (empty($properties['label'])) {
       throw new \InvalidArgumentException('The label property is required.');
     }
 
+    if (!empty($properties['is_admin']) && !is_bool($properties['is_admin'])) {
```

Comment:
I added it because this data is sourced from event subscribers, I thought it would help people making mistakes in their subscribers. I can remove it if you want.